presto-hive/src/main/java/com/facebook/presto/hive/orc/HdfsOrcDataSource.java

```diff
@@ -53,7 +54,11 @@ protected void readInternal(long position, byte[] buffer, int bufferOffset, int
             throw e;
         }
         catch (Exception e) {
-            throw new PrestoException(HIVE_FILESYSTEM_ERROR, format("HDFS error reading from %s at position %s", this, position), e);
+            String message = format("HDFS error reading from %s at position %s", this, position);
+            if (e.getClass().getSimpleName().equals("BlockMissingException")) {
```

Comment:
The only other place throws a different error code in the else case, so doesn't seem worth it to me. If we add more like this we can though